app/models/book.rb

```diff
@@ -3,6 +3,7 @@ class Book < ActiveRecord::Base
   has_many :order_items
 
   scope :by_published, -> { order(published_date: :desc) }
+  scope :most_popular, -> { all.includes(:order_items).sort_by(&:times_sold).reverse! }
```

Comment:
This smells bad to me. As it is, I believe it's iterating over all the items to sort them. If you use a counter_cache, you can have the DB do the heavy lifting for you.